sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -875,42 +886,92 @@ def download_manifest(self, repository, tag_or_digest, **kwargs):
             manifest_stream = _serialize_manifest(manifest)
+            if not _validate_digest(manifest_stream, digest):
+                raise ValueError("The requested digest does not match the digest of the received manifest.")
         except ValueError:
             if repository is None or tag_or_digest is None:
                 raise ValueError("The parameter repository and tag_or_digest cannot be None.")
-            if not _validate_digest(manifest_stream, digest):
-                raise ValueError("The requested digest does not match the digest of the received manifest.")
             raise
-
         return DownloadManifestResult(digest=digest, data=manifest_stream, manifest=manifest)
 
-    @distributed_trace
-    def download_blob(self, repository, digest, **kwargs):
-        # type: (str, str, **Any) -> Union[DownloadBlobResult, None]
-        """Download a blob that is part of an artifact.
+    @overload
+    def download_blob(self, repository: str, digest: str, **kwargs) -> DownloadBlobResult:
+        """Download a blob that is part of an artifact to memory.
 
-        :param str repository: Name of the repository
+        :param str repository: Name of the repository.
         :param str digest: The digest of the blob to download.
-        :returns: DownloadBlobResult or None
-        :rtype: ~azure.containerregistry.DownloadBlobResult or None
+        :returns: DownloadBlobResult
+        :rtype: ~azure.containerregistry.DownloadBlobResult
         :raises ValueError: If the parameter repository or digest is None.
         """
-        try:
-            deserialized = self._client.container_registry_blob.get_blob( # type: ignore
-                repository, digest, cls=_return_deserialized, **kwargs
-            )
-        except ValueError:
-            if repository is None or digest is None:
-                raise ValueError("The parameter repository and digest cannot be None.")
-            raise
 
-        if deserialized:
-            blob_content = b''
-            for chunk in deserialized:
-                if chunk:
-                    blob_content += chunk
-            return DownloadBlobResult(data=BytesIO(blob_content), digest=digest)
-        return None
+    @overload
+    def download_blob(self, repository: str, digest: str, destination: str, **kwargs) -> None:
+        """Download a blob that is part of an artifact to a file in destination.
+
+        :param str repository: Name of the repository.
+        :param str digest: The digest of the blob to download.
+        :param str destination: The destination of the blob download to.
+        :returns: None
+        :raises ValueError: If the parameter repository or digest is None.
+        """
+
+    @distributed_trace
+    def download_blob(self, *args: str, **kwargs) -> Union[DownloadBlobResult, None]:
+        repository = args[0]
+        digest = args[1]
+        if len(args) == 2:
```

Comment:
It seems that we're determining how to download based on the presence of the `destination` parameter.
What would happen if we _always_ use the `get_chunk` pattern rather than the `get_blob` pattern?

Reply:
Updated to always use get_chunk